ath79: add support for routerboard 960pgs

olemmela opened the pull request:
This PR adds support for a mikrotik 960pgs router known as hex poe.

olemmela committed:
ath79: add support for MikroTik RouterBOARD 960PGS (hEX PoE)

This patch adds support for the MikroTik RouterBOARD 960PGS (hEX PoE)
router. The device has a USB 2.0 port and a SFP port for adding optical
fiber connectivity. The ports 2-5 can power other PoE capable devices
with the same voltage as applied to the unit.

Specifications:

- SoC: Qualcomm Atheros QCA9557
- Flash: 16 MB (SPI)
- RAM: 128 MB
- 1x Ethernet SFP: 1000
- 1x Ethernet RJ45: 10/100/1000 port with passive POE in
- 4x Ethernet RJ45: 10/100/1000 ports with 802.3af/at PoE out
- 1x USB 2.0 host port
- 1x reset button

See https://mikrotik.com/product/RB960PGS for more details.

Flashing:
 TFTP boot initramfs image and then perform sysupgrade. Follow common
 MikroTik procedure as in https://openwrt.org/toh/mikrotik/common.

Signed-off-by: Oskari Lemmela <oskari@lemmela.net>

hauke commented on target/linux/ath79/mikrotik/config-default:
```diff
@@ -28,11 +31,15 @@ CONFIG_MTD_UBI_BLOCK=y
 CONFIG_MTD_UBI_BEB_LIMIT=20
 CONFIG_NET_DSA=y
+CONFIG_NET_DSA_QCA8K=y
+CONFIG_NET_DSA_TAG_QCA=y
```
I would prefer if we first activate CONFIG_NET_DSA_QCA8K for the compete target and not only for some boards.

olemmela replied:
commit c2140e32ce32b9cc60f7d408e20bdf45dce6a634 added new device with old switch driver. so also mikrotik subtarget now uses old switch driver.

hauke commented on target/linux/ath79/files/drivers/hwmon/rbpoeport.c:
```diff
@@ -0,0 +1,311 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
We already have multiple POE drivers in OpenWrt all with different interfaces. It is getting messy.

You could split this and we only take the board support without POE first.

f00b4r0 commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,310 @@
+	model = "MikroTik RouterBOARD 960PGS";
+
+	aliases {
+		serial0 = &uart;
```
Missing led aliases

olemmela replied:
I did miss led aliases completely. I'll add following aliases in next push
led-boot = &led_user;
led-failsafe = &led_user;
led-running = &led_user;
led-upgrade = &led_user;

f00b4r0 commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,310 @@
+				};
+
+				soft_config {
+					label = "soft_config";
```
Not needed

olemmela replied:
I'll remove this label

f00b4r0 commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,310 @@
+
+/ {
+	compatible = "mikrotik,routerboard-960pgs", "qca,qca9557";
+	model = "MikroTik RouterBOARD 960PGS";
```
maybe adjust to "MikroTik RouterBOARD 960PGS (hEx PoE)" as is done for other RB devices in tree

olemmela replied:
I'll change this in next push

f00b4r0 reviewed (changes requested):
I'm just pointing a few quick nits on the DTS for now.
Two quick remarks on the drivers code which I didn't review at this stage:
1) There seems to be some copyright info missing in the heading comments
2) the `rbpoe.h` header is GPL-2.0-only which is fine, the other files are GPL-3.0-or-later which is non-upstreamable (Linux kernel only accepts GPL-2.0-only).

walterav1984 commented:
RouterOS does support manual CPU frequency selection like 720/800/900/1000MHz, not sure if these settings are set by bootloader or nand variable... no mention of overclocking or warning whatsoever.

Is OpenWrt affected by these CPU settings or able to set them on its own, the default seems 800?

olemmela commented:
> RouterOS does support manual CPU frequency selection like 720/800/900/1000MHz, not sure if these settings are set by bootloader or nand variable... no mention of overclocking or warning whatsoever.
> 
> Is OpenWrt affected by these CPU settings or able to set them on its own, the default seems 800?

These are stored to soft_config. It is possible to modify them in openwrt too. Boot loader sets CPU freq and linux kernel does not modify it.

root@test:/sys/firmware/mikrotik/soft_config# cat cpufreq_index 
a b [c] d 
root@test:/sys/firmware/mikrotik/soft_config#

walterav1984 commented:
Will this router its switch, function without using commit [e98d67f](https://github.com/openwrt/openwrt/pull/9702/commits/e98d67f65ea358aad09a792242cc19d2627b85c1) and only commit [529e016](https://github.com/openwrt/openwrt/pull/9702/commits/529e016b9eec1df5e0e5a1dc5a1e814e3b18f65b) as in is QCA8K really needed?

> commit [c2140e3](https://github.com/openwrt/openwrt/commit/c2140e32ce32b9cc60f7d408e20bdf45dce6a634) added new device with old switch driver. so also mikrotik subtarget now uses old switch driver.

Is using the old switch driver still an option like this recently added [Mikrotik RouterBOARD hAP ac lite ](https://github.com/openwrt/openwrt/commit/2bd33e8626bd04fd7115ee1a42aaf03aae2fffb8) or is #4984 an acceptable workaround?

computeracer commented:
I am excited to see possible support for this platform. Please let me know if I can help with any testing.

piaddic7ed commented:
Thanks for adding this device.
I did some testing with your PR added to 23.05. 
The resulting initramfs boots fine, switch works, sfp works.
However, there are some minor things to which I'll get first and a major issue (at least for me).
A cosmetic issue, naming the device hEx instead of hEX which is the official writing.
For some reason the image contained wifi drivers which is not needed in my opinion.
So I changed this line a bit: `  DEVICE_PACKAGES += kmod-usb2 kmod-i2c-gpio kmod-sfp -iwinfo -swconfig \
       -kmod-ath9k -wpad-basic-mbedtls -libustream-mbedtls` 
The biggest issue, however, I can't get the flashed image to boot. The bootloader fails to load the image and keeps asking for netinstall/pxe boot. The RouterOS and firmware version was 7.11.
Any ideas?

f00b4r0 commented:
See https://forum.openwrt.org/t/mikrotik-routeros-v7-x-and-openwrt-sysupgrade/148072/6

piaddic7ed commented:
@f00b4r0 thanks for the hint. 
I ended up downgrading to 6.49.3 (the version I had handy).
Initramfs size limitations aside it was a smooth process. 
Boot via pxe, sysupgrade 23.05, done.
https://pastebin.com/P3hykF5R

olemmela commented:
> @olemmela Thanks for adding this device. I did some testing with your PR added to 23.05. The resulting initramfs boots fine, switch works, sfp works. However, there are some minor things to which I'll get first and a major issue (at least for me). A cosmetic issue, naming the device hEx instead of hEX which is the official writing. For some reason the image contained wifi drivers which is not needed in my opinion. So I changed this line a bit: ` DEVICE_PACKAGES += kmod-usb2 kmod-i2c-gpio kmod-sfp -iwinfo -swconfig \ -kmod-ath9k -wpad-basic-mbedtls -libustream-mbedtls` The biggest issue, however, I can't get the flashed image to boot. The bootloader fails to load the image and keeps asking for netinstall/pxe boot. The RouterOS and firmware version was 7.11. Any ideas?

Thanks for testing with 23.05 branch. My plan is rebase this PR to master after #13589 is merged.
Good point that wireless drivers can be dropped. I'll do that when rebasing.
Initramfs size limit is fixed in commit 62b72eafe49d2eecd36 so that should not be problem anymore.

Upstream kernel have basic netlink based POE support nowadays. It should be quite simple task to rewrite
POE driver to use that.

I don't currently have any spare devices to test. All of them are in remote locations. So poe driver
testing is more than welcome.

olemmela commented:
@piaddic7ed  if you want to test rebased version https://github.com/olemmela/openwrt/tree/rb960pgs-rebase
For the switch led support, testing 6.1 kernel is needed.

olemmela committed:
ath79: mikrotik: convert net drivers as modules

Change net, phy and change drivers to modules.
This change allows you to build a device image without these drivers.

Signed-off-by: Oskari Lemmela <oskari@lemmela.net>

tonve commented:
another rb960 there, built from
https://github.com/olemmela/openwrt/tree/rb960pgs-rebase
dmesg
https://pastebin.com/EmwzZVkr
All leds are ok. I have no poe hw, passive poe may be tested with dumb soho voltage tester.

AreYouLoco commented:
Please add RB960PGS-PB (PowerBox Pro) as variant. Its basically the same device hardware-wise but different box for outdoor usage.

hauke commented on target/linux/ath79/mikrotik/config-default:
```diff
@@ -1,3 +1,8 @@
+CONFIG_AG71XX_LEGACY=n
+CONFIG_AG71XX_LEGACY_DEBUG_FS=n
+CONFIG_AR8216_PHY=n
+CONFIG_AR8216_PHY_LEDS=n
+CONFIG_AT803X_PHY=n
```
Please use:
```
# CONFIG_AT803X_PHY is not set
```

Please also run this to refresh the configuration:
```
make kernel_oldconfig CONFIG_TARGET=subtarget
```

hauke commented:
I would prefer if you first only add the device and we can handle the DSA switch separately.
Is the DSA driver mandatory for some features of this device?

The ath79 DSA driver still shows some performance problems. 

When we build the switch drivers as a module the total image gets bigger, please post the sizes of some of the images before and after this change.

olemmela commented:
> I would prefer if you first only add the device and we can handle the DSA switch separately. Is the DSA driver mandatory for some features of this device?
> 

The new pse-pd framework uses netlink to control POE and each POE port must be a separate ethernet device. The POE part has been left out of this PR, but it could be added as its own PR after this is merged.

> The ath79 DSA driver still shows some performance problems.
> 

The performance issues are related to GRO because the ag71xx driver does not support HW checksum. Disabling GRO on all interfaces resolves the performance issues.

> When we build the switch drivers as a module the total image gets bigger, please post the sizes of some of the images before and after this change.

It looks like the image sizes actually got a bit smaller
`$ diff -u1 old_sizes.txt new_sizes.txt 
--- old_sizes.txt	2025-05-06 21:14:43.569134169 +0300
+++ new_sizes.txt	2025-05-06 21:21:11.340822863 +0300
@@ -1,62 +1,62 @@
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-493g-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-493g-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-493g-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-493g-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-750-r2-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-750-r2-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-750-r2-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-5hpacd-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-5hpacd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-5hpacd-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-5hpacd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-xhpnd-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-xhpnd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-xhpnd-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-911g-xhpnd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-911-lite-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-911-lite-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-911-lite-squashfs-sysupgrade.bin
 
-6948 openwrt-ath79-mikrotik-mikrotik_routerboard-912uag-2hpnd-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-912uag-2hpnd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-912uag-2hpnd-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-912uag-2hpnd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-921gs-5hpacd-15s-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-921gs-5hpacd-15s-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-921gs-5hpacd-15s-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-921gs-5hpacd-15s-squashfs-sysupgrade.bin
 
-6948 openwrt-ath79-mikrotik-mikrotik_routerboard-922uags-5hpacd-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-922uags-5hpacd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-922uags-5hpacd-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-922uags-5hpacd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-951g-2hnd-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-951g-2hnd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-951g-2hnd-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-951g-2hnd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2hnd-initramfs-kernel.bin
-7372 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2hnd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2hnd-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2hnd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-951ui-2nd-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-952ui-5ac2nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-952ui-5ac2nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-952ui-5ac2nd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-962uigs-5hact2hnt-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-962uigs-5hact2hnt-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-962uigs-5hact2hnt-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-lhg-2nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-lhg-2nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-lhg-2nd-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-lhg-5nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-lhg-5nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-lhg-5nd-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-map-2nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-map-2nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-map-2nd-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-mapl-2nd-initramfs-kernel.bin
-7432 openwrt-ath79-mikrotik-mikrotik_routerboard-mapl-2nd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-mapl-2nd-initramfs-kernel.bin
+7428 openwrt-ath79-mikrotik-mikrotik_routerboard-mapl-2nd-squashfs-sysupgrade.bin
 
-6948 openwrt-ath79-mikrotik-mikrotik_routerboard-sxt-5nd-r2-initramfs-kernel.bin
-7376 openwrt-ath79-mikrotik-mikrotik_routerboard-sxt-5nd-r2-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-sxt-5nd-r2-initramfs-kernel.bin
+7344 openwrt-ath79-mikrotik-mikrotik_routerboard-sxt-5nd-r2-squashfs-sysupgrade.bin
 
-6944 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-2nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-2nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-2nd-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-g-5hact2hnd-initramfs-kernel.bin
-7432 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-g-5hact2hnd-squashfs-sysupgrade.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-g-5hact2hnd-initramfs-kernel.bin
+7428 openwrt-ath79-mikrotik-mikrotik_routerboard-wap-g-5hact2hnd-squashfs-sysupgrade.bin
 
-6940 openwrt-ath79-mikrotik-mikrotik_routerboard-wapr-2nd-initramfs-kernel.bin
+6916 openwrt-ath79-mikrotik-mikrotik_routerboard-wapr-2nd-initramfs-kernel.bin
 7428 openwrt-ath79-mikrotik-mikrotik_routerboard-wapr-2nd-squashfs-sysupgrade.bin
`

hauke commented on target/linux/ath79/image/mikrotik.mk:
```diff
@@ -4,7 +4,7 @@ define Device/mikrotik_routerboard-493g
   SOC := ar7161
   DEVICE_MODEL := RouterBOARD 493G
-  DEVICE_PACKAGES += kmod-usb-ohci kmod-usb2
+  DEVICE_PACKAGES += kmod-usb-ohci kmod-usb2 kmod-switch-ar8xxx kmod-swconfig
```
You do not have to specify `kmod-swconfig`, it will be automatically added because `kmod-switch-ar8xxx` depends on it.

hauke commented:
LGTM.

Is GRO deactivated on the ports by default? 

@Ansuel  what do you think? Should we use DSA for this device?

olemmela commented:
> LGTM.
> 
> Is GRO deactivated on the ports by default?
> 

Currently GRO is enabled by default. The user can disable it manually or another option would be to modify the ag71xx driver to disable it by default. Or add a default configuration that disables GRO on all ports.

DragonBluep commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,343 @@
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
```
```suggestion
```

DragonBluep commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,343 @@
+	};
+
+	i2c: i2c {
+		compatible = "i2c-gpio";
```
Please add:
```
			#address-cells = <1>;
			#size-cells = <0>;
```

olemmela replied:
changed

DragonBluep commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,343 @@
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		led_sfp: sfp {
```
```suggestion
		sfp {
```

DragonBluep commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,343 @@
+				reg = <0>;
+				label = "cpu";
+				ethernet = <&eth0>;
+				phy-mode = "rgmii-id";
```
It's better to insert one empty line.

DragonBluep commented on target/linux/ath79/image/mikrotik.mk:
```diff
@@ -4,7 +4,8 @@ define Device/mikrotik_routerboard-2011uias-2hnd
   DEVICE_MODEL := RouterBOARD 2011UiAS-2HnD
-  DEVICE_PACKAGES += kmod-usb2 kmod-i2c-gpio kmod-sfp kmod-gpio-beeper
+  DEVICE_PACKAGES += kmod-usb2 kmod-i2c-gpio kmod-sfp kmod-gpio-beeper \
+	kmod-switch-ar8xxx
```
`kmod-switch-ar8xxx` is an empty package on ath79 target. We don't need to add it.

DragonBluep commented on target/linux/ath79/mikrotik/base-files/etc/board.d/02_network:
```diff
@@ -47,6 +47,9 @@ ath79_setup_interfaces()
 		ucidef_set_interface_lan "eth0 eth1"
 		;;
+	mikrotik,routerboard-960pgs)
+		ucidef_set_interfaces_lan_wan "lan2 lan3 lan4 lan5" "sfp lan1"
```
Maybe move lan1 to lan group by default?

olemmela replied:
Port labeling is bit misleading see https://cdn.mikrotik.com/web-assets/rb_images/1220_xl.webp
They are labeled as 1 to 5 where port 1 and sfp ports are wan ports.

DragonBluep commented on target/linux/ath79/mikrotik/config-default:
```diff
@@ -1,3 +1,6 @@
+# CONFIG_AG71XX_LEGACY is not set
+# CONFIG_AR8216_PHY is not set
+# CONFIG_AT803X_PHY is not set
```
Please do this in a separated PR so that this PR can be accepted as soon as possible.

DragonBluep commented on target/linux/ath79/mikrotik/target.mk:
```diff
@@ -3,7 +3,7 @@ FEATURES += minor nand
 IMAGES_DIR := ../../..
 
-DEFAULT_PACKAGES += wpad-basic-mbedtls yafut
+DEFAULT_PACKAGES += wpad-basic-mbedtls yafut kmod-ag71xx-legacy kmod-phy-at803x
```
Please do this in a separated PR.

olemmela replied:
done #21031

DragonBluep commented on target/linux/ath79/dts/qca9557_mikrotik_routerboard-960pgs.dts:
```diff
@@ -0,0 +1,343 @@
+
+	reg_usb_vbus {
+		compatible = "regulator-fixed";
+		regulator-always-on;
```
Please add `regulator-boot-on;`.

DragonBluep commented on target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds:
```diff
@@ -42,6 +42,13 @@ mikrotik,routerboard-952ui-5ac2nd)
+	ucidef_set_led_netdev "lan2" "LAN2" "qca8k-0.0:01:green:lan" "lan2" "link-10 link-100 link-1000 tx rx"
+	ucidef_set_led_netdev "lan3" "LAN3" "qca8k-0.0:02:green:lan" "lan3" "link-10 link-100 link-1000 tx rx"
+	ucidef_set_led_netdev "lan4" "LAN4" "qca8k-0.0:03:green:lan" "lan4" "link-10 link-100 link-1000 tx rx"
+	ucidef_set_led_netdev "lan5" "LAN5" "qca8k-0.0:04:green:lan" "lan5" "link-10 link-100 link-1000 tx rx"
```
Do we really need these configs? These LEDs should have some default triggers.

dangowrt commented:
> device-tree is used for describing hardware

A label printed on a case *is* still hardware. The property has been suggested upstream for our downstream use, see https://lore.kernel.org/netdev/20240709124503.pubki5nwjfbedhhy@skbuf/

arinc9 commented:
> > device-tree is used for describing hardware
> 
> A label printed on a case *is* still hardware.

Who said otherwise?

Djfe commented:
was just merged feel free to close
https://github.com/openwrt/openwrt/commit/078fab3397924d06699034ca744dc5e5670ce045
